pepper-apis/src/test/java/org/broadinstitute/ddp/service/AgeUpServiceTest.java

```diff
@@ -174,6 +189,58 @@ public void testRunAgeUpCheck_reachedAgeOfMajorityPrep() {
         });
     }
 
+    @Test
+    public void testRunAgeUpCheck_errorInOneCandidateDoesNotAffectAnother() {
+        TransactionWrapper.useTxn(handle -> {
+            User user1 = createAgeUpTestCandidate(handle, EnrollmentStatusType.REGISTERED, LocalDate.of(2000, 1, 14));
+            User user2 = createAgeUpTestCandidate(handle, EnrollmentStatusType.REGISTERED, LocalDate.of(2000, 1, 24));
+            handle.attach(InvitationFactory.class)
+                    .createInvitation(InvitationType.AGE_UP, testData.getStudyId(), user1.getId(), "test@datadonationplatform.org");
+
+            // Setup downstream event which will fail for second candidate
+            long triggerId = handle.attach(EventTriggerDao.class)
+                    .insertStaticTrigger(EventTriggerType.REACHED_AOM_PREP);
+            long actionId = handle.attach(EventActionDao.class)
+                    .insertInvitationEmailNotificationAction(new SendgridEmailEventActionDto("template", "en"));
+            handle.attach(JdbiEventConfiguration.class).insert(triggerId, actionId, testData.getStudyId(),
+                    Instant.now().toEpochMilli(), null, 0, null, null, true, 1);
+
+            GovernancePolicy policy = new GovernancePolicy(1L, testData.getStudyId(), testData.getStudyGuid(), new Expression("true"));
+            policy.addAgeOfMajorityRule(new AgeOfMajorityRule("true", 20, 4));
+            assertEquals("should have only completed 1 candidate",
+                    1, service.runAgeUpCheck(handle, interpreter, policy));
+
+            try {
+                // Wait for events to be ready
+                TimeUnit.SECONDS.sleep(1);
+            } catch (InterruptedException ignored) {
+            }
```

Comment:
You might want to log this,  as an interruption here might lead to a test failure.